huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection_associate.go

```diff
@@ -0,0 +1,236 @@
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The region where the datasource enhanced connection and elastic resource pools are located.`,
+			},
+			"connection_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The ID of the datasource enhanced connection to be associated.`,
+			},
+			"elastic_resource_pools": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `The list of the elastic resource pool names.`,
+			},
+		},
+	}
+}
+
+func buildCreateDatasourceConnectionAssociateBodyParams(poolNames *schema.Set) map[string]interface{} {
+	return map[string]interface{}{
+		"elastic_resource_pools": utils.ExpandToStringListBySet(poolNames),
+	}
+}
+
+func associateResourcePoolsToConnection(client *golangsdk.ServiceClient, connectionId string, poolNames *schema.Set) error {
+	httpUrl := "v2.0/{project_id}/datasource/enhanced-connections/{connection_id}/associate-queue"
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createPath = strings.ReplaceAll(createPath, "{connection_id}", connectionId)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildCreateDatasourceConnectionAssociateBodyParams(poolNames)),
+	}
+
+	requestResp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return fmt.Errorf("error associating the elastic resource pools to the enhanced connection: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return err
+	}
+	if !utils.PathSearch("is_success", respBody, true).(bool) {
+		return fmt.Errorf("unable to associate the elastic resource pools to the enhanced connection: %s",
+			utils.PathSearch("message", respBody, "Message Not Found"))
+	}
+	return nil
+}
+
+func resourceDatasourceConnectionAssociateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg          = meta.(*config.Config)
+		region       = cfg.GetRegion(d)
+		connectionId = d.Get("connection_id").(string)
+	)
+
+	client, err := cfg.NewServiceClient("dli", region)
+	if err != nil {
+		return diag.Errorf("error creating DLI client: %s", err)
+	}
+
+	err = associateResourcePoolsToConnection(client, connectionId, d.Get("elastic_resource_pools").(*schema.Set))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(connectionId)
+
+	return resourceDatasourceConnectionAssociateRead(ctx, d, meta)
+}
+
+func GetDatasourceConnectionAssociatedPools(client *golangsdk.ServiceClient, connectionId string) ([]interface{}, error) {
+	respBody, err := getConnectionById(client, connectionId)
+	if err != nil {
+		return nil, err
+	}
+
+	associatedPools := utils.PathSearch("elastic_resource_pools", respBody, make([]interface{}, 0)).([]interface{})
```

Comment:
how about directly search the names

Reply:
yeah, should be "elastic_resource_pools[*].name"